client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -146,7 +146,10 @@ class PreviewFrame extends React.Component {
   injectLocalFiles() {
     const htmlFile = this.props.htmlFile.content;
     let scriptOffs = [];
-    const resolvedFiles = this.resolveJSAndCSSLinks(this.props.files);
+    const files = this.props.files.slice();
```

Comment:
could you move this into a little helper function? i know this function is already long and messy, so i'm hoping to not make it worse 😈

Reply:
Updated! Sorry for the turnaround time. LMK if this isn't the right approach.